app/assets/javascripts/host_power.js

```diff
@@ -0,0 +1,64 @@
+(function(){
+
+  $(document).ready(function () {
+    powerStatus();
+  });
+
+  function hostIds() {
+    var res = [];
+
+    $('#hosts-table tbody tr').each( function (index, item) {
+      res.push($(item).data("host-id"));
+    });
+
+    return res;
+  }
+
+  function findSpinner(id) {
+    return $('#hosts-table tbody').find("tr[data-host-id='" + id + "']").find(".spinner");
+  }
+
+  function updatePowerStatus(response) {
+    var spinner = null;
+    hostIds().forEach(function (item) {
+      spinner = findSpinner(item);
+      if ((typeof response !== "string") && response[item]) {
```

Comment:
Perhaps you should add a `find_multiple_ajax` method then that returns a json with an error status in case of an error instead of redirecting back to the index?